### PR TITLE
Update the Open Union implementation to OpenUnion51.hs

### PR DESCRIPTION
**Pull request by joshvera**
Also changes the `Member` and `Members` Open Union type classes to `:<` and `:<:`.

/cc @robrix

### CODE REVIEW

File: bench/Core.hs [hunk 1]
```diff
@@ -58,16 +58,16 @@ data Http out where
...
-close' :: Member Http r => Eff r ()
+close' :: (Http :< r) => Eff r ()
 close' = send Close
 
-post' :: Member Http r => String -> Eff r String
+post' :: (Http :< r) => String -> Eff r String
 post' = send . Post
 
-get' :: Member Http r => Eff r String
+get' :: (Http :< r) => Eff r String
```
Comment (robrix) [inline bench/Core.hs]: We could enable `UnicodeSyntax` and then use `∈` 😆

Comment (robrix) [inline bench/Core.hs]: This shouldn’t conflict with `Cofree` because it’s of kind `Constraint`, right?

Comment (joshvera) [inline bench/Core.hs]: Yep, unfortunately type operators have to be parenthesized in the constraint part of the declaration. `:<` and `:<:` seemed reasonable, but I’m open to operator suggestions!

Comment (robrix) [inline bench/Core.hs]: These seem fine to me if `Member`/`Members` are undesirable.

**Pull request merged**